### PR TITLE
track calls

### PR DESCRIPTION
**Pull request by anoonan**
This PR adds functionality for track calls. It breaks out `order completed` since appboy has a specific method for that:

- order completed: https://js.appboycdn.com/web-sdk/latest/doc/module-appboy.html#.logPurchase
- all other track calls: https://js.appboycdn.com/web-sdk/latest/doc/module-appboy.html#.logCustomEvent

cc @hankim813 @tsholmes

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -66,31 +69,51 @@ Appboy.prototype.loaded = function() {
...
+  var currencyCode = track.currency();
+  var purchaseProperties = track.properties();
+
+  // remove reduntant properties
+  del(purchaseProperties, 'products');
+  del(purchaseProperties, 'currency');
+
+  // call appboylogPurcahse for each product
+  each(function(product) {
+    var track = new Track({ properties: product });
```
Comment (hankim813) [inline lib/index.js:113]: ayyyy i 👀 u with these moves

File: test/index.test.js [hunk 2]
```diff
@@ -60,24 +60,82 @@ describe('Appboy', function() {
...
       });
 
-      it('should send the name of the event', function() {
-        analytics.track('event name');
-        analytics.called();
+      it('should send send all properties', function() {
+        analytics.track('event with properties', {
+          nickname: 'noonz',
+          spiritAnimal: 'wolf',
+          best_friend: 'tom'
```
Comment (hankim813) [inline test/index.test.js:78]: wow

File: test/index.test.js [hunk 3]
```diff
@@ -60,24 +60,82 @@ describe('Appboy', function() {
...
+        });
+        analytics.called(window.appboy.logCustomEvent, 'event with properties', {
+          nickname: 'noonz',
+          spiritAnimal: 'wolf',
+          best_friend: 'tom'
```
Comment (hankim813) [inline test/index.test.js:83]: at least make this one han

**Review by hankim813 — APPROVED** [review]

File: lib/index.js (hunk 1 above)
Comment (anoonan) [reply lib/index.js:113]: hair_flip

**Pull request merged**